# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositorEGL.cpp
# Commit: 6841dc516087
# Full Hash: 6841dc516087239f3e48806a38356ad725c2c272
# Author: stransky <stransky@redhat.com>
# Date: 2022-09-21 21:43:38
# Regressor Bug: 1777664
# File Overlap Count: 1
# Description:
#   Bug 1645677 [Wayland] Update EGLSurface when wl_surface is deleted r=emilio,jgilbert
#   
#   When GtkWidget is hidden, underlying wl_surface is deleted. We need to also update EGLSurface of GtkWidget (GtkCompositorWidget)
#   as EGLSurface is directly linked to wl_surface:
#   
# ==============================================================================

diff -r 5ba3a56b3038 -r 6841dc516087 gfx/webrender_bindings/RenderCompositorEGL.cpp
--- a/gfx/webrender_bindings/RenderCompositorEGL.cpp	Wed Sep 21 10:32:44 2022 +0000
+++ b/gfx/webrender_bindings/RenderCompositorEGL.cpp	Wed Sep 21 10:32:44 2022 +0000
@@ -127,6 +127,11 @@
 #endif
 
   RenderedFrameId frameId = GetNextRenderFrameId();
+#ifdef MOZ_WAYLAND
+  if (mWidget->IsHidden()) {
+    return frameId;
+  }
+#endif
   if (mEGLSurface != EGL_NO_SURFACE && aDirtyRects.Length() > 0) {
     gfx::IntRegion bufferInvalid;
     const auto bufferSize = GetBufferSize();